torchvision/csrc/cpu/video_reader/FfmpegStream.cpp

```diff
@@ -0,0 +1,267 @@
+#include "FfmpegStream.h"
+#include "FfmpegUtil.h"
+
+using namespace std;
+
+// (TODO) Currently, disable the use of refCount
+static int refCount = 0;
+
+FfmpegStream::FfmpegStream(
+    AVFormatContext* inputCtx,
+    int index,
+    enum AVMediaType avMediaType,
+    double seekFrameMargin)
+    : inputCtx_(inputCtx),
+      index_(index),
+      avMediaType_(avMediaType),
+      seekFrameMargin_(seekFrameMargin) {
+}
+
+FfmpegStream::~FfmpegStream() {
+  if (frame_) {
+    av_frame_free(&frame_);
+  }
+  avcodec_free_context(&codecCtx_);
+}
+
+int FfmpegStream::openCodecContext() {
+  VLOG(2) << "stream start_time: " << inputCtx_->streams[index_]->start_time;
+
+  auto typeString = av_get_media_type_string(avMediaType_);
+  AVStream* st = inputCtx_->streams[index_];
+  auto codec_id = st->codecpar->codec_id;
+  VLOG(1) << "codec_id: " << codec_id;
+  AVCodec* codec = avcodec_find_decoder(codec_id);
+  if (!codec) {
+    LOG(ERROR) << "avcodec_find_decoder failed for codec_id: " << int(codec_id);
+    return AVERROR(EINVAL);
+  }
+  VLOG(1) << "Succeed to find decoder";
+
+  codecCtx_ = avcodec_alloc_context3(codec);
+  if (!codecCtx_) {
+    LOG(ERROR) << "avcodec_alloc_context3 fails";
+    return AVERROR(ENOMEM);
+  }
+
+  int ret;
+  /* Copy codec parameters from input stream to output codec context */
+  if ((ret = avcodec_parameters_to_context(codecCtx_, st->codecpar)) < 0) {
+    LOG(ERROR) << "Failed to copy " << typeString
+               << " codec parameters to decoder context";
+    return ret;
+  }
+
+  AVDictionary* opts = nullptr;
+  av_dict_set(&opts, "refcounted_frames", refCount ? "1" : "0", 0);
+
+  // after avcodec_open2, value of codecCtx_->time_base is NOT meaningful
+  // But inputCtx_->streams[index_]->time_base has meaningful values
+  if ((ret = avcodec_open2(codecCtx_, codec, &opts)) < 0) {
+    LOG(ERROR) << "avcodec_open2 failed. " << ffmpeg_util::getErrorDesc(ret);
+    return ret;
+  }
+  VLOG(1) << "Succeed to open codec";
+
+  frame_ = av_frame_alloc();
+  return initFormat();
+}
+
+unique_ptr<DecodedFrame> FfmpegStream::getFrameData(int getPtsOnly) {
+  if (!codecCtx_) {
+    LOG(ERROR) << "Codec is not initialized";
+    return nullptr;
+  }
+  if (getPtsOnly) {
+    unique_ptr<DecodedFrame> decodedFrame = make_unique<DecodedFrame>();
+    decodedFrame->pts_ = frame_->pts;
+    return decodedFrame;
+  } else {
+    unique_ptr<DecodedFrame> decodedFrame = sampleFrameData();
+    if (decodedFrame) {
+      decodedFrame->pts_ = frame_->pts;
+    }
+    return decodedFrame;
+  }
+}
+
+<<<<<<< Updated upstream
```

Comment:
It seems that you forgot some merge conflicts in this file

Reply:
Hey Lowik, it is fixed in the replacement PR (https://github.com/pytorch/vision/pull/1303)